core/save-profiledata.c

```diff
@@ -0,0 +1,204 @@
+	put_int(b, entry->stopdepth_calc);
+	put_int(b, entry->pressure_time);
+	put_int(b, entry->heartbeat);
+	put_int(b, entry->bearing);
+	put_double(b, entry->ambpressure);
+	put_double(b, entry->gfline);
+	put_double(b, entry->surface_gf);
+	put_double(b, entry->density);
+	put_int(b, entry->icd_warning ? 1 : 0);
+}
+
+
+void put_headers(struct membuffer *b)
+{
+	put_csv_string(b, "in_deco");
+	put_csv_string(b, "sec");
+	for (int c = 0; c < MAX_CYLINDERS; c++) {
+		put_format(b, "\"pressure_%d_cylinder\", ", c);
+		put_format(b, "\"pressure_%d_interpolated\", ", c);
+	}
+	put_csv_string(b, "temperature");
+	put_csv_string(b, "depth");
+	put_csv_string(b, "ceiling");
+	for (int i = 0; i < 16; i++) {
+		put_format(b, "\"ceiling_%d\", ", i);
+	}
+	for (int i = 0; i < 16; i++) {
+		put_format(b, "\"percentage_%d\", ", i);
+	}
+	put_csv_string(b, "ndl");
+	put_csv_string(b, "tts");
+	put_csv_string(b, "rbt");
+	put_csv_string(b, "stoptime");
+	put_csv_string(b, "stopdepth");
+	put_csv_string(b, "cns");
+	put_csv_string(b, "smoothed");
+	put_csv_string(b, "sac");
+	put_csv_string(b, "running_sum");
+	put_csv_string(b, "pressureo2");
+	put_csv_string(b, "pressuren2");
+	put_csv_string(b, "pressurehe");
+	put_csv_string(b, "o2pressure");
+	put_csv_string(b, "o2sensor0");
+	put_csv_string(b, "o2sensor1");
+	put_csv_string(b, "o2sensor2");
+	put_csv_string(b, "o2setpoint");
+	put_csv_string(b, "scr_oc_po2");
+	put_csv_string(b, "mod");
+	put_csv_string(b, "ead");
+	put_csv_string(b, "end");
+	put_csv_string(b, "eadd");
+	put_csv_string(b, "velocity");
+	put_csv_string(b, "speed");
+	put_csv_string(b, "in_deco_calc");
+	put_csv_string(b, "ndl_calc");
+	put_csv_string(b, "tts_calc");
+	put_csv_string(b, "stoptime_calc");
+	put_csv_string(b, "stopdepth_calc");
+	put_csv_string(b, "pressure_time");
+	put_csv_string(b, "heartbeat");
+	put_csv_string(b, "bearing");
+	put_csv_string(b, "ambpressure");
+	put_csv_string(b, "gfline");
+	put_csv_string(b, "surface_gf");
+	put_csv_string(b, "density");
+	put_csv_string(b, "icd_warning");
+}
+
+void save_profiles_buffer(struct membuffer *b, bool select_only)
+{
+	int i;
+	struct dive *dive;
+	struct plot_info pi;
+	struct deco_state *planner_deco_state = NULL;
+
+	for_each_dive(i, dive) {
+		if (select_only && !dive->selected)
+			continue;
+		pi = calculate_max_limits_new(dive, &dive->dc);
+		create_plot_info_new(dive, &dive->dc, &pi, false, planner_deco_state);
+		put_headers(b);
+		put_format(b, "\n");
+
+		for (int i = 0; i < pi.nr; i++) {
+			put_pd(b, &pi.entry[i]);
+			put_format(b, "\n");
+		}
+
+		put_format(b, "\n");
+
+	}
+}
+
+int save_profiledata(const char *filename, const bool select_only)
+{
+	struct membuffer buf = { 0 };
+	FILE *f;
+	int error = 0;
+
+
+	save_profiles_buffer(&buf, select_only);
+
+	if (same_string(filename, "-")) {
+		f = stdout;
+	} else {
+		f = subsurface_fopen(filename, "w");
+	}
+	if (f) {
+		flush_buffer(&buf, f);
+		error = fclose(f);
+	}
+	if (error)
```

Comment:
I'm not sure I understand the logic here:
If `subsurface_open()` fails, error stays 0 and therefore no error is produced. Or am I missing something?

Reply:
Honestly, this was copy-pasted from save_dives_logic() in save_xml.c

Reply:
Ah, no, I deleted too much.